### ISSUE TITLE
Input disable with Chinese IME starting from QON windows-b3623

### ISSUE THREAD

**Issue by vanleo2001** [pbek/QOwnNotes#976]
Today I update QON, then I find I can't imput Chinese character with Chinese IME. 
I test the previous build version. I can input Chinese character with Chinese IME if QON <= 18.05.7 ( windows-b3618 ), but can't  if QON >= 18.06.0 ( windows-b3623 ), even updated 18.07.4 ( windows-b3709 ). I guess this error may also appear with Japanese IME or Korean IME.

**Comment by pbek** [comment]
The main thing that might have changed in that time was the Qt version of the Windows build. Can you please post the debug information (see issue template) of the two versions when you run them?

**Comment by pbek** [comment]
Are there any messages in the log (see issue template)?

**Comment by abev66** [comment]
My fcitx doesn't work sometimes. But if I switch to other note and switch back, it will be work again. :\

OS: Manjaro Linux 
IME: fcitx 4.2.9.5
Qt: 5.11.1

**Comment by vanleo2001** [comment]
I test again. Just as @abev66 said, when I run QON and open first note, it can't use Chinese IME, but if I switch other note, Chinese IME can be used. Also I cover the QOwnNotes.exe with windows-b3623 version while other librarys are from windows-b3618, it is same as above. So I guess it is not caused by library updating.

**Comment by piyo-gh** [comment]
Well, I didn't use QOwnNotes for a long time, but tried to test with today's version.

1. Start QOwnNotes.
2. Press F4.
IME is disabled.
But as already mentioned, if I switch to other note and switch back, it will be work again.

An another case to confirm the problem.

1. Press F4 and edit with IME ON. <- This is OK.
2. Press Alt+Shift+E and press Alt+Shift+E again.
IME is disabled.

'Disabled' means current control might not be recognized as editable control.

**Comment by pbek** [comment]
Thank you for the feedback, I'm not sure how I even should reproduce this issue, yet alone find out what causes it.

**Comment by vanleo2001** [comment]
With the help of debugger, I find when QOwnNotes.exe start running in windows build 3715, the default first opening note will be set as readonly. 
https://github.com/pbek/QOwnNotes/blob/9140281c534e693ef222a5caa3b8abc35f0a5927/src/mainwindow.cpp#L10120

https://github.com/pbek/QOwnNotes/blob/9140281c534e693ef222a5caa3b8abc35f0a5927/src/mainwindow.cpp#L10124

```
void MainWindow::updateNoteTextEditReadOnly() {
    setNoteTextEditReadOnly(
            !(currentNote.exists() && currentNote.fileWriteable() &&
              Utils::Misc::isNoteEditingAllowed()));

    if (ui->noteTextEdit->isVisible() && currentNote.hasEncryptedNoteText()) {
        ui->noteTextEdit->setReadOnly(true);
    }
}
```
Stepping ``currentNote.exists() `` will return FALSE. 
If I manually set it TRUE in debugger, I can input with Chinese IME.

It The below old version, you just set roughly as **unchecked**, so no problem.
https://github.com/pbek/QOwnNotes/blob/270241250b26f23c9778bf0d37927b849cc501f7/src/mainwindow.cpp#L9950

**Comment by pbek** [comment]
@vanleo2001 thank you for the debugging on your system. So for you the whole note was set to read-only and you couldn't type at all? I thought this only is a IME issue.

Does anyone else can't type at all?

**Comment by vanleo2001** [comment]
No. It is so strange that I can input English charactor with Chinese IME, can't input Chinese charactor with Chinese IME.

**Comment by pbek** [comment]
So your IME thinks the textedit is read-only, even if it isn't?
Does this also happen with other text-edits in QOwnNotes?

**Comment by vanleo2001** [comment]
Just for the first auto opened note.
In debugger, it shows currentNote.exists()=false for the first auto opened note, currentNote.exists()=true for other notes not auto opened first

**Comment by pbek** [comment]
> In debugger, it shows currentNote.exists()=false for the first auto opened note

Are you sure that really was the first note that was shown on the screen? Or just a nil-note because no note was selected yet and then the last opened note was selected.

You shouldn't be able to see and edit the note with your keyboard if your note would be shown with `currentNote.exists()=false`.

**Comment by vanleo2001** [comment]
I'm not sure if the "first note" in debugger is a nil-note. I'm sure the problem is here https://github.com/pbek/QOwnNotes/blob/9140281c534e693ef222a5caa3b8abc35f0a5927/src/mainwindow.cpp#L10124

**Comment by pbek** [comment]
I'm still not sure what I should do about it. If you are able to edit the notes with the keyboard then it looks to me like a problem of IME isn't detecting if the text-edit is read-only or not.

**Comment by vanleo2001** [comment]
Rollback https://github.com/pbek/QOwnNotes/blob/9140281c534e693ef222a5caa3b8abc35f0a5927/src/mainwindow.cpp#L10124
to ``  setNoteTextEditReadOnly(!checked); `` ?

**Comment by pbek** [comment]
> Rollback

That would break the read-only mode button.

**Comment by pbek** [comment]
So far I wasn't even able to get Fcitx to work to get some Chinese or Japanese input method so I'm still in the dark...

**Comment by vanleo2001** [comment]
Is the QT5 bug?  Ref: https://bugreports.qt.io/browse/QTBUG-55699

**Comment by vanleo2001** [comment]
Please review this QT5 bug:  https://bugreports.qt.io/browse/QTBUG-55699
I build and run the [Attachments](https://bugreports.qt.io/secure/attachment/58637/qtbug55699.zip) in this bug report. After click the button to set QPlainTextEdit  read-only flag as True , then set as False, Japanese IME can only input English charactor. 
This is same as QON. When the default first note opens, QPlainTextEdit  has set read-only TWICE (First is False, Second isTrue). So this is a QT5 bug, it affects Version/s: 5.6.1, 5.7.0, and not fix

**Comment by pbek** [comment]
Hey, awesome. Great job in finding a solution, @vanleo2001! If https://codereview.qt-project.org/#/c/169997/3/src/widgets/widgets/qplaintextedit.cpp really fixes the problem maybe I can override the method for QOwnNotes!

**Comment by pbek** [comment]
## 18.07.7
- attempted to fix a problem with Chinese and Japanese input methods

**Comment by pbek** [comment]
There now is a new release, could you please test it and report if it works for you?

**Comment by pbek** [comment]
@piyo-gh and @abev66 could you please also test the new release?

**Comment by vanleo2001** [comment]
I test, the bug fixed.

**Comment by abev66** [comment]
Tested +1 
It fixed.

**Comment by pbek** [comment]
Phew, I'm very glad. Thank you for all the help. I hope fcitx works too for @piyo-gh.

**Comment by pbek** [comment]
Awesome, thank you!

**Comment by pbek** [comment]
https://bugreports.qt.io/browse/QTBUG-55699 should be fixed in Qt 5.12 too now.
